docs/DeveloperGuide.md

```diff
@@ -133,89 +133,188 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Undo/Redo Feature
 
-#### Proposed Implementation
+The Undo/Redo feature allows users to revert wrongly executed commands.
 
-The proposed undo/redo mechanism is facilitated by `VersionedAddressBook`. It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+This section explains the implementation of the Undo and Redo mechanism and highlights the design considerations taken into account when implementing this feature.
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+#### Implementation
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+The Undo/Redo mechanism is designed around maintaining a history of `TutorsPet` states, and restoring a particular states when the user triggers an undo or redo command.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+![TutorsPetState](images/TutorsPetState.png)
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+A `TutorsPetState` contains a Tutor's Pet state, along with a message that describes the changes relevant to this state.
```

Comment:
Not sure if this would sound better...
```suggestion
A `TutorsPetState` object contains a Tutor's Pet state, along with a message that describes the changes relevant to this state.
```